src/lib/CollectStore.js

```diff
@@ -127,41 +123,34 @@ export default class CollectStore {
   // https://github.com/cozy/cozy-stack/blob/master/docs/konnectors_workflow_example.md
   connectAccount(konnector, account, disableEnqueue, enqueueAfter = 10000) {
     // return object to store all business object implied in the connection
-    const connection = {}
+    const connection = {
+      konnector: konnector
+    }
     // detect oauth case
     const isOAuth = !!account.oauth
 
-    // 1. Konnector installation
+    // 1. Create folder, will be replaced by an intent or something else
     return (
-      konnectors
-        .install(cozy.client, konnector, INSTALL_TIMEOUT)
-        .then(completeKonnector => {
-          connection.konnector = { ...konnector, ...completeKonnector }
-        })
-        // 2. Create folder, will be replaced by an intent or something else
-        .then(() =>
-          this.createDirectoryIfNecessary(
-            !!account.auth && account.auth.folderPath
-          ).then(folder => {
-            if (!folder) return Promise.resolve()
-            connection.folder = folder
-            // 3. Reference konnector in folder
-            return konnectors
-              .addFolderPermission(
-                cozy.client,
+      this.createDirectoryIfNecessary(!!account.auth && account.auth.folderPath)
+        .then(folder => {
+          if (!folder) return Promise.resolve()
+          connection.folder = folder
+          // 2. Reference konnector in folder
+          return konnectors
+            .addFolderPermission(
+              cozy.client,
+              connection.konnector,
+              connection.folder._id
+            )
+            .then(permission => {
+              connection.permission = permission
+              return cozy.client.data.addReferencedFiles(
                 connection.konnector,
                 connection.folder._id
               )
-              .then(permission => {
-                connection.permission = permission
-                return cozy.client.data.addReferencedFiles(
-                  connection.konnector,
-                  connection.folder._id
-                )
-              })
-          })
-        )
-        // 4. Creates account
+            })
+        })
+        // 3. Creates account
```

Comment:
Should create the trigger before the accounts? Is there a risk to get orphan accounts here?

Reply:
Unfortunately we need the account to create the trigger.